src/mongo/MongoScrapbook.ts

```diff
@@ -70,6 +73,8 @@ export function getAllCommandsFromTextDocument(document: vscode.TextDocument): M
 }
 
 async function executeCommands(context: IActionContext, commands: MongoCommand[]): Promise<void> {
+    clearResults = true;
+    isExecutingAll = true;
```

Comment:
What if you call `openReadOnlyContent` here and pass in `ReadOnlyContent` as an optional parameter to `executeCommand`. That way you know you always open a fresh read-only editor before you start an "Execute All" and I think that would allow you to get rid of all your file-scoped variables above